spec/forms/support_interface/application_forms/pick_course_form_spec.rb

```diff
@@ -62,5 +77,15 @@
   describe 'validations' do
     it { is_expected.to validate_presence_of(:course_option_id) }
     it { is_expected.to validate_presence_of(:application_form_id) }
+
+    it 'checks that the course is open on Apply' do
+      course_scope = class_double(Course)
```

Comment:
I was initially wondering if we can create a course here instead of using doubles, but then with testing found that running it as so without any courses still passes:
```
it 'checks that the course is open on Apply' do
    form = described_class.new(course_option_id: '123', course_code: 'ABC')
    expect(form.valid?(:save)).to be false
    expect(form.errors[:course_option_id]).to eq(['This course is not open on the Apply service'])
end
```

I was wondering if the intention was to also include non existing course ids being passed in with this error, or if we should handle that separately with a different error, let me know your thoughts

Reply:
Ah yes, so we're not currently validating the existence of the course for the given `course_option_id` param in the update action presumably on the assumption that a support user is not going to craft data here.
If we think this validation is useful (I'm wondering if it is given it's the support UI) we could just make the validation error message more generic, or have 2 validations. either way it's very unlikely to ever be seen given we are excluding unopen courses from the choices that can be submitted here.

Reply:
Thats a good point, I think we can keep it as is. 👍 

My only comment now is around preferring the creation of the course rather than using doubles, as that is tying our test to the internal implementation, and I think creation of the course isn't very expensive, what do you think?

Reply:
I intentionally avoided creating a db record in the interests of speed, I'm happy to use `create` if that's preferred. We've typically tried to reduce db record creation for the sake of the overall speed of the test suite.

Reply:
@cwrw I've removed the double as this wasn't necessary. The spec works fine using `FactoryBot#build` for the course model so we still save a trip to the db.